### DISCUSSION TITLE
Add support for custom headers in redirects.

### DISCUSSION THREAD

**Discussion by vorillaz** [vercel/next.js#38838] [Ideas]
### Describe the feature you'd like to request

Upon guarding an SSR route with redirects it's pretty common to redirect users back to the login page if there the users are unauthenticated, sometimes though we might need to control the headers for a specific redirect manually.
Moreover that approach could also work as communication channel upon requests.

### Describe the solution you'd like

An API that allows overriding headers upon redirecting:
```js
module.exports = {
  async redirects() {
    return [
      {
        source: '/about',
        statusCode: 301,
        headers: {
            "x-redirect": "foo", 
       }
      },
    ]
  },
}

```

### Describe alternatives you've considered

I am not sure that there will is a really constructive and clear alternative.

**Comment by daniel-nagy** [comment]
I just wrote a blog post about how you can [add response headers to redirects in Vercel](https://danielnagy.me/posts/Post_pqf05clj4hyr). I'm not using Next.js but it should be similar, you can just use `NextResponse.next` and `NextResponse.redirect` to make things easier. It would be nice if you could just include headers in the configuration though. Surprised this isn't supported.
